src/classes/service-proxy/service-proxy.cpp

```diff
@@ -0,0 +1,73 @@
+#include "service-proxy.hpp"
+
+auto Status::getState() -> Status::stateT
+{
+
+    const std::lock_guard<std::mutex> lock(m_mtx);
+
+    return m_state;
+}
+
+void Status::setState(Status::stateT newState)
+{
+
+    const std::lock_guard<std::mutex> lock(m_mtx);
+
+    m_state = newState;
+}
+
+void ServiceProxy::servicePod(IService& service, Status& status)
+{
+
+    service.setup();
+    status.setState(Status::RUNNING);
+
+    while (status.getState() != Status::STOPED) {
+
+        service.routine();
+    }
+
+    service.destroy();
+}
+
+ServiceProxy::ServiceProxy(IService& service, nlohmann::json configs)
+    : m_innerService(service)
+{
+    m_dependencies.data = configs; //DETAILS ABOUT DEPENDENCIES
+
+    m_status.setState(Status::UNINITIALIZED);
+    if (configs["dependencies"] == 0) {
+        run();
+    }
+}
+
+ServiceProxy::~ServiceProxy()
+{
+    auto currState = m_status.getState();
+    if (currState == Status::RUNNING) {
+        stop();
+    }
+}
+
+void ServiceProxy::run()
+{
+    std::thread thread(servicePod, std::ref(m_innerService), std::ref(m_status));
```

Comment:
Tente fazer isso
```suggestion
    std::thread thread(&ServiceProxy::servicePod, this);
```
https://stackoverflow.com/questions/17472827/create-thread-inside-class-with-function-from-same-class